### PR TITLE
Add documentation describing required setup and how to run EtcdProxyController

### PR DESCRIPTION
**Pull request by xmudrii**
This PR adds documentation describing setup needed to run EtcdProxyController and how to run EtcdProxyController. The documentation also covers how to run the core etcd using the provided sample manifests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,26 +6,81 @@ Implements https://groups.google.com/forum/#!msg/kubernetes-sig-api-machinery/rH
...
 
 HEAD of this repo matches versions `1.10` of `k8s.io/apiserver`, `k8s.io/apimachinery`, and `k8s.io/client-go`.
 
-## Running
+## Running controller
 
-Prerequisite: Since the sample-controller uses apps/v1 deployments, the Kubernetes cluster version should be greater than 1.9.
+There're two ways to run `EtcdProxyController`:
 
+* in-cluster, which is done by deploying [the deployment manifest, located in the `artifacts/deployment` directory](https://github.com/xmudrii/etcdproxy-controller/blob/master/artifacts/deployment/00-etcdproxy-controller.yaml),
```
Comment (sttts) [inline README.md]: does a relative path work? This might break and point to a wrong branch/tag.

File: README.md [hunk 2]
```diff
@@ -6,26 +6,81 @@ Implements https://groups.google.com/forum/#!msg/kubernetes-sig-api-machinery/rH
...
+
+The deployment manifest assumes etcd is available on `https://etcd-svc-1.etcd.svc:2379`. This can be configured by updating the [`00-etcdproxy-controller.yaml` manifest](https://github.com/xmudrii/etcdproxy-controller/blob/master/artifacts/deployment/00-etcdproxy-controller.yaml).
+
+When running out-of-cluster, the etcd URL can be changed using the `etcd-core-url` flag.
+
+### Providing the core etcd client certificates to the controller
+
+When deploying the core etcd using the provided manifests, you can deploy the client certificates using the `etcd-client-certs.yaml` manifest. [The README of `artifacts/etcd`](https://github.com/xmudrii/etcdproxy-controller/tree/master/artifacts/etcd) contains more details about deploying the client certificates.
+
+If you're deploying etcd using another method, the `EtcdProxyController` requires you to provide it the CA certificate (`ca.pem`) as a ConfigMap, and the client certificate (`client.pem`) and key (`client-key.pem`) as
```
Comment (sttts) [inline README.md]: what's their name?

Comment (sttts) [reply README.md]: nvmd, describe 5 lines below :D

Comment (sttts) [reply README.md]: maybe put the "in the controller namespace" into parenthesis in this paragraph and put a colon behind. Then it is clear that the precise description follows.

File: README.md [hunk 3]
```diff
@@ -6,26 +6,81 @@ Implements https://groups.google.com/forum/#!msg/kubernetes-sig-api-machinery/rH
...
-# assumes you have a working kubeconfig, not required if operating in-cluster
-$ go run *.go -kubeconfig=$HOME/.kube/config
 
-# create a CustomResourceDefinition
-$ kubectl create -f artifacts/examples/etcdstorage-crd.yaml
+### Running out-of-cluster
 
-# create a custom resource of type Foo
-$ kubectl create -f artifacts/examples/etcdstorage-cr.yaml
+Before running the controller out-of-cluster you need to create the etcd proxy namespace, which is by default called `kube-apiserver-storage`, and can be configured using the `--namespace` flag:
```
Comment (sttts) [inline README.md]: better: by default called `...` (can be customized using `--namespace`):

File: README.md [hunk 4]
```diff
@@ -6,26 +6,81 @@ Implements https://groups.google.com/forum/#!msg/kubernetes-sig-api-machinery/rH
...
+### Providing the core etcd client certificates to the controller
+
+When deploying the core etcd using the provided manifests, you can deploy the client certificates using the `etcd-client-certs.yaml` manifest. [The README of `artifacts/etcd`](https://github.com/xmudrii/etcdproxy-controller/tree/master/artifacts/etcd) contains more details about deploying the client certificates.
+
+If you're deploying etcd using another method, the `EtcdProxyController` requires you to provide it the CA certificate (`ca.pem`) as a ConfigMap, and the client certificate (`client.pem`) and key (`client-key.pem`) as
+a Secret.
+
+Currently, both the ConfigMap and Secret must be located in the controller namespace.
+
+The ConfigMap is by default called `etcd-coreserving-ca`, and the name can be changed using the `--etcd-core-ca-configmap` flag.
```
Comment (sttts) [inline README.md]: why is called `... coreserving ... `? Sounds confusing. `etcd-core-ca` and `etcd-core-client-cert` would make more sense, no?

Comment (xmudrii) [reply README.md]: Makes sense. As I'm trying to focus this PR on docs-only changes, let's leave it for another PR if that's okay. We can also discuss this on-call today.

File: README.md (hunk 1 above)
Comment (xmudrii) [reply README.md]: Updated all URLs to use relative paths.

File: README.md (hunk 3 above)
Comment (xmudrii) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (xmudrii) [reply README.md]: Updated.

File: README.md [hunk 5]
```diff
@@ -6,26 +6,81 @@ Implements https://groups.google.com/forum/#!msg/kubernetes-sig-api-machinery/rH
...
 
 ## Compatibility
 
 HEAD of this repo matches versions `1.10` of `k8s.io/apiserver`, `k8s.io/apimachinery`, and `k8s.io/client-go`.
 
-## Running
+## Running controller
 
-Prerequisite: Since the sample-controller uses apps/v1 deployments, the Kubernetes cluster version should be greater than 1.9.
+There're two ways to run `EtcdProxyController`:
```
Comment (deads2k) [inline README.md]: There are

**Comment by deads2k** [comment]
When someone has difficulty (someone trying it will always find something), be responsive to fix it.  This is moving in the right direction.

**Pull request merged**